pkg/daemon/ceph/osd/daemon.go

```diff
@@ -66,11 +66,9 @@ func Provision(context *clusterd.Context, agent *OsdAgent) error {
 	// set the crush location in the osd config file
 	cephConfig := mon.CreateDefaultCephConfig(context, agent.cluster, path.Join(context.ConfigDir, agent.cluster.Name))
 	cephConfig.GlobalConfig.CrushLocation = agent.location
-	// don't set public or cluster addr if prepare only
-	if agent.prepareOnly {
-		cephConfig.PublicAddr = ""
-		cephConfig.ClusterAddr = ""
-	}
+	cephConfig.PublicAddr = ""
```

Comment:
Could you readd a comment why these two vars need to be empty

Reply:
@rootfs do you recall why the prepare pod couldn't set the public and cluster addr?

Reply:
i'm going to try removing this, i can't find a need for it.